File: Leetcode_POTD_2025/8_August/1493_Longest_Subarray_of_1s_After_Deleting_One_Element.py
class Solution(object):
    def longestSubarray(self, nums):
        z_c = 0
        l_p, m_l = 0, 0
        for r_p, n in enumerate(nums):
            if n == 0:
                z_c += 1
            while z_c > 1:
                if nums[l_p] == 0:
                    z_c -= 1
                l_p += 1
            m_l = max(m_l, r_p - l_p)
        return m_l

File: Leetcode_POTD_2025/8_August/1493_Longest_Subarray_of_1s_After_Deleting_One_Element.cpp
class Solution {
public:
    int longestSubarray(vector<int>& nums) {
        int z_c = 0, l_p = 0, m_l = 0;
        for (int r_p = 0; r_p < nums.size(); r_p++) {
            if (nums[r_p] == 0) z_c++;
            while (z_c > 1) {
                if (nums[l_p] == 0) z_c--;
                l_p++;
            }
            m_l = max(m_l, r_p - l_p);
        }
        return m_l;
    }
};
